### PR TITLE
[tlse] TLS database connection

### PR DESCRIPTION
**Pull request by stuggi**
The my.cnf file gets added to the secret holding the service configs. The content of my.cnf is centrally managed in the mariadb-operator and retrieved calling db.GetDatabaseClientConfig(tlsCfg)

Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/190
Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/191

Jira: [OSPRH-4547](https://issues.redhat.com//browse/OSPRH-4547)

### CODE REVIEW

File: controllers/ironicneutronagent_controller.go [hunk 1]
```diff
@@ -647,11 +648,22 @@ func (r *IronicNeutronAgentReconciler) generateServiceConfigMaps(
 
 	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(ironic.ServiceName), map[string]string{})
 
+	db, err := mariadbv1.GetDatabaseByName(ctx, h, ironic.DatabaseName)
```
Comment (stuggi) [inline controllers/ironicneutronagent_controller.go:651]: @hjensas do the neutronagent even need/has access to the db?

Comment (hjensas) [reply controllers/ironicneutronagent_controller.go:651]: No direct DB access in the agent.

Comment (stuggi) [reply controllers/ironicneutronagent_controller.go]: ack, thanks. I have reverted the change to the agent. please have a look when you get to it. right now there is an issue in ci.

**Review by hjensas — APPROVED** [review]
Thanks Martin! 

LGTM

File: templates/common/bin/common.sh [hunk 2]
```diff
@@ -86,7 +86,7 @@ function common_ironic_config {
         crudini --set ${SVC_CFG_MERGED} DEFAULT transport_url $TRANSPORTURL
         crudini --set ${SVC_CFG_MERGED} DEFAULT rpc_transport oslo
     fi
-    crudini --set ${SVC_CFG_MERGED} database connection mysql+pymysql://${DBUSER}:${DBPASSWORD}@${DBHOST}/${DB}
+    crudini --set ${SVC_CFG_MERGED} database connection mysql+pymysql://${DBUSER}:${DBPASSWORD}@${DBHOST}/${DB}?read_default_file=/etc/my.cnf
```
Comment (zzzeek) [inline templates/common/bin/common.sh:89]: this needed to be added in ironicinspector/bin/init.sh also, and the TLS volume mounts added to ironicinspector/dbsync.go .  ill fix this in my mariadbaccount patch

**Pull request merged**